### DISCUSSION TITLE
global .gitAuthor isn't being used

### DISCUSSION THREAD

**Discussion by codezninja** [renovatebot/renovate#11546] [Request Help] [answered]
I noticed that my renovate has started opening up issues cause it couldn't push to the repos. Reading the debug logs. It seems like my self-hosted `gitAuthor` isn't being used anymore in each of the repos. I'm trying to understand this [PR](https://github.com/renovatebot/renovate/pull/11539)

Does this mean that global gitAuthor in `OrgName/.github/default.json` in self-hosted repo is no longer respected and we now need to add this value on each renovate.json per repo?

**Comment by rarkins** [comment]
The net result should be that configuring `gitAuthor` is now more flexible, but we may have had a bug in an earlier release.

Which exact Renovate version did you run?

**Reply by codezninja** [reply]
So I'm using the docker tag I was using `25-slim` and just updated `26-slim`. That version seems to be `26.16.0`

**Reply by rarkins** [reply]
Please check it for sure from your logs

**Reply by codezninja** [reply]
yup the logs are debug and they show `"renovateVersion": "26.16.0"`

**Reply by codezninja** [reply]
![image](https://user-images.githubusercontent.com/4400773/131874065-045a28ca-f5a3-414d-8d81-668dedf1010c.png)
I confirmed by downloading the latest docker image as well

**Reply by rarkins** [reply]
Please describe exactly how you're using that preset with gitAuthor so I can understand the use case

**Reply by rarkins** [reply]
The PR you pointed to was released in 26.17.0 so I got confused. Please upgrade to 26.17.0 when it's available as I think it will resolve you problem

**Reply by codezninja** [reply]
so I have two files in my local org `.github/default.json` & `.github/renvoate-config.json`

default.json contains the following information
```
{
    "extends": [
      "config:base",
      ":disableRateLimiting"
    ],
    "gitAuthor": "username <user@company.com>",
    "reviewers": [
      "team:developers"
    ]
}
```

renovate-config.json
```
{
  "extends": [
    "local>OrgName/.github"
  ]
}
```


The app repo contains the following `renovate.json`
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "local>OrgName/.github:renovate-config"
  ]
}
```

**Reply by codezninja** [reply]
thanks yeah I'll wait for the docker image to be available. Is there a build process I can watch to see when it'll be pushed?

**Reply by rarkins** [reply]
It will be built here (main branch): https://github.com/renovatebot/docker-renovate/actions

**Reply by codezninja** [reply]
seems like now gitAuthor works. My workflow picked up  "renovateVersion": "26.17.1" and now its able to push. there's some other error messages for some other things but I think thats the version change. I think once I get the latest patch version it should fix all my issues.

Thanks for helping me debug

**Comment by rarkins** [comment]
> Does this mean that global gitAuthor in `OrgName/.github/default.json` in self-hosted repo is no longer respected and we now need to add this value on each renovate.json per repo?

No, that's not what's intended

**Comment by rarkins** [comment]
Can you also describe your use case in more detail? e.g.
- Hosted app or self-hotsed?
- If self-hosted, were you configuring custom git author in your global (bot) config, or part of user/repo config using presets?

**Reply by codezninja** [reply]
so this is self-hosted on GHE.

I had configured gitAuthor in `.github/default.json`. I have a `.github/renovate-config.json` which extends `"local>OrgName/.github"` which is pulling that default.json

**Comment by codezninja** [comment]
For a little more context these is the error messages that seems to have opened the issue. This could be cause I don't have the latest version of renovate docker image avaiable to download from docker hub. Maybe we can see if I can use the latest to see if the issue persists.

```
Author identity unknown\n\n*** Please tell me who you are.\n\nRun\n\n  git config --global user.email \"you@example.com\"\n  git config --global user.name \"Your Name\"\n\nto set your account's default identity.\nOmit --global to set the identity only in this repository.\n\nfatal: unable to auto-detect email address (got 'ubuntu@02c9cbc8f268.(none)')\n",
         "stack": "Error: Author identity unknown\n\n*** Please tell me who you are.\n\nRun\n\n  git config --global user.email \"you@example.com\"\n  git config --global user.name \"Your Name\"\n\nto set your account's default identity.\nOmit --global to set the identity only in this repository.\n\nfatal: unable to auto-detect email address (got 'ubuntu@02c9cbc8f268.(none)')\n\n    at Object.action (/usr/src/app/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:38:28)\n    at PluginStore.exec (/usr/src/app/node_modules/simple-git/src/lib/plugins/plugin-store.ts:24:29)\n    at /usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:114:40\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:111:14)\n    at GitExecutorChain.<anonymous> (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:88:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.js:5:58)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)"
```

**Reply by viceice** [reply]
Please provide some more context debug logs

**Comment by viceice** [comment]
You never set `gitAuthor` in bot admin / global config? `gitAuthor` from preset would never be used before renovate [v26.17.0](https://github.com/renovatebot/renovate/releases/tag/26.17.0)

So maybe you forgot to set `gitAuthor` in bot config? It's required now if renovate couldn't detect it from the platform.

**Reply by rarkins** [reply]
It must be related to the changes I made today, and should be resolved in the next release

**Reply by viceice** [reply]
@rarkins https://github.com/renovatebot/renovate/pull/11536 I think it's because you removed the fallback when no author could be fetched via api.

**Reply by codezninja** [reply]
seems like I was using the fallback. Cause another issue had cropped up after 26.17.1. 

Using latest docker images `26.17.1` & `26.18.0` my previous already onboarded repos all worked fine now back to where I only set gitAuthor in default.json. But new repos I wanted to onboarded now complained with the following errors.

```
INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 50,
           "logContext": "_KCPpedwT",
           "repository": "OrgName/.github",
           "err": {
             "task": {
               "commands": ["commit", "-m", "Add renovate.json", "--no-verify"],
               "format": "utf-8"
             },
             "message": "Author identity unknown\n\n*** Please tell me who you are.\n\nRun\n\n  git config --global user.email \"you@example.com\"\n  git config --global user.name \"Your Name\"\n\nto set your account's default identity.\nOmit --global to set the identity only in this repository.\n\nfatal: unable to auto-detect email address (got 'ubuntu@d00c6632b6d4.(none)')\n",
             "stack": "Error: Author identity unknown\n\n*** Please tell me who you are.\n\nRun\n\n  git config --global user.email \"you@example.com\"\n  git config --global user.name \"Your Name\"\n\nto set your account's default identity.\nOmit --global to set the identity only in this repository.\n\nfatal: unable to auto-detect email address (got 'ubuntu@d00c6632b6d4.(none)')\n\n    at Object.action (/usr/src/app/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:38:28)\n    at PluginStore.exec (/usr/src/app/node_modules/simple-git/src/lib/plugins/plugin-store.ts:24:29)\n    at /usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:114:40\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:111:14)\n    at GitExecutorChain.<anonymous> (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:88:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.js:5:58)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)"
           },
           "msg": "Repository has unknown error"
         }
       ]
```
 
But I fixed the issue by passing in these additional variables into the docker container

```
GIT_AUTHOR_NAME
GIT_AUTHOR_EMAIL
GIT_COMMITTER_NAME
GIT_COMMITTER_EMAIL
```

So based on your help. I've removed gitAuthor from default.json preset file and now in the docker container I pass all these variables

```
RENOVATE_GIT_AUTHOR
GIT_AUTHOR_NAME
GIT_AUTHOR_EMAIL
GIT_COMMITTER_NAME
GIT_COMMITTER_EMAIL
```

**Reply by viceice** [reply]
`RENOVATE_GIT_AUTHOR` should be the only required environment variable.

In general renovate should be able to learn this from the platform.

**Comment by dahaupt** [comment, marked as answer]
We experienced the same issue with `v26.16.0`. With the latest tag and version `26.17.1` it is working again, using the correct platform `gitAuthor`.

**Reply by codezninja** [reply]
@dahaupt curious do you have an issue when onboarding a new repo? 26.17.1 seems to work for existing repos but when I try to onboard a new one I'm getting an error.

```
stack": "Error: Author identity unknown\n\n*** Please tell me who you are.\n\nRun\n\n  git config --global user.email \"you@example.com\"\n  git config --global user.name \"Your Name\"\n\nto set your account's default identity.\nOmit --global to set the identity only in this repository.\n\nfatal: unable to auto-detect email address (got 'ubuntu@28cca93853d6.(none)')\n\n    at Object.action (/usr/src/app/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:38:28)\n    at PluginStore.exec (/usr/src/app/node_modules/simple-git/src/lib/plugins/plugin-store.ts:24:29)\n    at /usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:114:40\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:111:14)\n    at GitExecutorChain.<anonymous> (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:88:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.js:5:58)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)"
```

**Reply by codezninja** [reply]
from what I can tell these are now required if running renovate in a container for at least onboarding
```
GIT_AUTHOR_NAME
GIT_AUTHOR_EMAIL
GIT_COMMITTER_NAME
GIT_COMMITTER_EMAIL
```

**Reply by rarkins** [reply]
No, that should not be the case

**Reply by rarkins** [reply]
Which platform are you using?

**Reply by codezninja** [reply]
The repos are all in Github Enterprise, I'm running renovate as a github action using a docker container.

**Reply by viceice** [reply]
You should then use our official GitHub action.

**Reply by codezninja** [reply]
We can't use the github markeplace actions yet, so till then I've been running it with a docker container.

**Reply by rarkins** [reply]
I've been able to reproduce the onboarding commit problem now and will either fix or revert within the next hour

**Comment by marverix** [comment]
We are experiencing the same issue. We are using Self Hosted version `35-slim`. We have configured `gitAuthor`. The platform is `bitbucket`.

Log:
```
08:23:36               "message": "Committer identity unknown\n\n*** Please tell me who you are.\n\nRun\n\n  git config --global user.email \"you@example.com\"\n  git config --global user.name \"Your Name\"\n\nto set your account's default identity.\nOmit --global to set the identity only in this repository.\n\nfatal: unable to auto-detect email address (got 'ubuntu@8a85b182daa1.(none)')\n",
08:23:36               "stack": "Error: Committer identity unknown\n\n*** Please tell me who you are.\n\nRun\n\n  git config --global user.email \"you@example.com\"\n  git config --global user.name \"Your Name\"\n\nto set your account's default identity.\nOmit --global to set the identity only in this repository.\n\nfatal: unable to auto-detect email address (got 'ubuntu@8a85b182daa1.(none)')\n\n    at Object.action (/opt/buildpack/tools/renovate/35.96.0/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:42:29)\n    at PluginStore.exec (/opt/buildpack/tools/renovate/35.96.0/node_modules/simple-git/src/lib/plugins/plugin-store.ts:29:29)\n    at /opt/buildpack/tools/renovate/35.96.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:127:42\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/opt/buildpack/tools/renovate/35.96.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:124:14)\n    at GitExecutorChain.<anonymous> (/opt/buildpack/tools/renovate/35.96.0/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:100:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/opt/buildpack/tools/renovate/35.96.0/node_modules/simple-git/dist/cjs/index.js:55:24)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)"
```

The funny fact is that it is working "randomly". In ~85% it works correctly. For example, it will go thru 20 repos and will work, and on 21 will fail. In the next run it will fail right away. In the next it will not fail at all. I don't see any pattern there.

**Reply by rarkins** [reply]
Please start a new discussion. It's not the same problem and you will need to provide logs of working and non working scenarios
